client/test/router_store.spec.js

```diff
@@ -0,0 +1,669 @@
+          lastPage: {
+            info: 'last-page',
+            isComplete: false,
+          },
+          result: {},
+          timestamp: new Date(),
+        },
+      });
+      assert.isTrue(routerStore.hasNextPage());
+      dispatcher.dispatch({
+        type: ActionTypes.MESSAGE_FETCH_SUCCESS,
+        value: {
+          lastPage: {
+            info: 'last-page',
+            isComplete: true,
+          },
+          result: {},
+          timestamp: new Date(),
+        },
+      });
+      assert.isFalse(routerStore.hasNextPage());
+    });
+    it('getLastPage', () => {
+      loadMessages([fixtures.message1, fixtures.message2]);
+      assert.deepEqual(routerStore.getLastPage(), fixtures.lastPage);
+    });
+    it('isPageComplete', () => {
+      changeRoute({ });
+      loadMessages([fixtures.message1, fixtures.message2]);
+      routerStore.getMessagesList();
+      assert.isFalse(routerStore.isPageComplete());
+      const msgs = generateMessages();
+      loadMessages(msgs);
+      routerStore.getMessagesList();
+      assert.isTrue(routerStore.isPageComplete());
+    });
+    it('getMessagesList', () => {
+      const msgs = [];
+      for (let i = 0; i < UtilConstants.MSGBYPAGE + 3; i++) {
+        const message = _.clone(fixtures.message1);
+        message.id = `id${i}`;
+        message.messageID = `meid${i}`;
+        message.conversationID = `c${i}`;
+        msgs.push(message);
+      }
+      msgs[12].flags = [];
+      msgs[14].flags = [MessageFlags.FLAGGED];
+      msgs[15].flags = [MessageFlags.FLAGGED, MessageFlags.ATTACH];
+      changeRoute({ });
+      loadMessages(msgs);
+      assert.equal(routerStore.getMessagesList().size, msgs.length);
+      changeRoute({ flags: MessageFilter.UNSEEN });
+      assert.equal(routerStore.getMessagesList().size, 3);
+      changeRoute({ flags: MessageFilter.FLAGGED });
+      assert.equal(routerStore.getMessagesList().size, 2);
+      changeRoute({ flags: MessageFilter.ATTACH });
+      assert.equal(routerStore.getMessagesList().size, 1);
+    });
+    it('getConversation', () => {
+      const msgs = generateMessages();
+      msgs[16].conversationID = 'c5';
+      msgs[17].conversationID = 'c5';
+      msgs[18].conversationID = 'c5';
+
+      changeRoute({ });
+      loadMessages(msgs);
+      assert.equal(routerStore.getConversation('c5').length, 4);
+    });
+    it('getConversationLength', () => {
+      const msgs = generateMessages();
+      changeRoute({ });
+      loadMessages(msgs, { c5: 6 });
+      assert.equal(routerStore.getConversationLength({
+        conversationID: 'c5',
+      }), 6);
+    });
+    it.skip('getNextConversation', () => {
+      const msgs = generateMessages();
+      changeRoute({ });
+      msgs[2].conversationID = 'c1';
+      msgs[3].conversationID = 'c1';
+      assert.equal(routerStore.getNextConversation().get('messageID'),
+                   msgs[3].messageID);
+    });
+    it.skip('getPreviousConversation', () => {
```

Comment:
Following our IRC conversation, I will try to handle it. If problem comes from the store code, I will write tests, then comment it.